File: scripts/fetch-products.mjs
// scripts/fetch-products.mjs
// Récupère les produits Amazon via Creators API et génère data.jsx.
//
// Usage:
//   node scripts/fetch-products.mjs            → fait toutes les catégories
//   node scripts/fetch-products.mjs --test     → test rapide avec une seule recherche
//
// Variables d'env requises :
//   AMAZON_CREATORS_CREDENTIAL_ID
//   AMAZON_CREATORS_CREDENTIAL_SECRET
//   AMAZON_CREATORS_CREDENTIAL_VERSION (3.2 pour l'Europe)
//   AMAZON_PARTNER_TAG

import { readFile, writeFile } from "node:fs/promises";
import { join } from "node:path";

// ────────────────────────────────────────────────────────────────────────────
// Charge d'abord l'environnement du processus, puis les .env locaux disponibles.
async function loadEnv() {
    const env = { ...process.env };
    for (const envPath of [join(process.cwd(), ".env"), "/Users/Yann/Documents/oracle/bonsplansmania/.env"]) {
        try {
            const raw = await readFile(envPath, "utf-8");
            for (const line of raw.split("\n")) {
                const m = line.match(/^([A-Z_][A-Z0-9_]*)=(.*)$/);
                if (m && !env[m[1]]) env[m[1]] = m[2].replace(/^["']|["']$/g, "");
            }
        } catch {
            // Fichier facultatif : les secrets CI viennent de l'environnement.
        }
    }
    return env;
}

function creatorsConfig(env) {
    const version = env.AMAZON_CREATORS_CREDENTIAL_VERSION || "3.2";
    const tokenUrl = version.startsWith("3.")
        ? "https://api.amazon.co.uk/auth/o2/token"
        : "https://creatorsapi.auth.eu-south-2.amazoncognito.com/oauth2/token";
    return { version, tokenUrl };
}

async function getAccessToken(env) {
    const id = env.AMAZON_CREATORS_CREDENTIAL_ID;
    const secret = env.AMAZON_CREATORS_CREDENTIAL_SECRET;
    const { version, tokenUrl } = creatorsConfig(env);
    const isV3 = version.startsWith("3.");
    const headers = { "content-type": isV3 ? "application/json" : "application/x-www-form-urlencoded" };
    const body = isV3
        ? JSON.stringify({ grant_type: "client_credentials", client_id: id, client_secret: secret, scope: "creatorsapi::default" })
        : new URLSearchParams({ grant_type: "client_credentials", client_id: id, client_secret: secret, scope: "creatorsapi/default" });
    const res = await fetch(tokenUrl, { method: "POST", headers, body });
    const data = await res.json();
    if (!res.ok || !data.access_token) throw new Error(`Creators API auth ${res.status}: ${JSON.stringify(data).slice(0, 300)}`);
    return { token: data.access_token, version };
}

async function searchItems(env, auth, { keywords, searchIndex = "All", itemCount = 10 }) {
    const payload = {
        partnerTag: env.AMAZON_PARTNER_TAG,
        marketplace: env.AMAZON_MARKETPLACE || "www.amazon.fr",
        keywords,
        searchIndex,
        itemCount,
        resources: [
            "images.primary.large",
            "itemInfo.title",
            "itemInfo.byLineInfo",
            "offersV2.listings.price",
            "offersV2.listings.availability",
        ],
    };
    const authorization = auth.version.startsWith("2.")
        ? `Bearer ${auth.token}, Version ${auth.version}`
        : `Bearer ${auth.token}`;
    const res = await fetch("https://creatorsapi.amazon/catalog/v1/searchItems", {
        method: "POST",
        headers: { authorization, "content-type": "application/json", "x-marketplace": payload.marketplace },
        body: JSON.stringify(payload),
    });
    const text = await res.text();
    if (!res.ok) {
        throw new Error(`Creators API ${res.status}: ${text.slice(0, 500)}`);
    }
    return JSON.parse(text);
}

// ────────────────────────────────────────────────────────────────────────────
async function main() {
    const env = await loadEnv();
    if (!env.AMAZON_CREATORS_CREDENTIAL_ID || !env.AMAZON_CREATORS_CREDENTIAL_SECRET) {
        console.error("❌ Identifiants Creators API manquants. Créez-les dans Associates Central > Outils > Creators API.");
        process.exit(1);
    }
    if (!env.AMAZON_PARTNER_TAG) throw new Error("AMAZON_PARTNER_TAG manquant");
    const auth = await getAccessToken(env);
    console.log("🔑 Creators API connectée · Tag:", env.AMAZON_PARTNER_TAG, "· Version:", auth.version);

    if (process.argv.includes("--test")) {
        console.log("\n🧪 Test : recherche 'soutien-gorge dentelle'");
        const data = await searchItems(env, auth, { keywords: "soutien-gorge dentelle", searchIndex: "Apparel", itemCount: 3 });
        const items = data?.searchResult?.items || [];
        console.log(`✅ API OK — ${items.length} produits reçus`);
        for (const it of items.slice(0, 3)) {
            console.log(`  · ${it.asin} — ${(it.itemInfo?.title?.displayValue || "").slice(0, 70)} — ${it.offersV2?.listings?.[0]?.price?.money?.displayAmount || "—"}`);
        }
        return;
    }

    // Catégories Maison Léa
    const CATEGORIES = [
        { id: "lingerie", keywords: ["ensemble dentelle femme", "soutien-gorge balconnet", "body dentelle", "bralette sans armature", "culotte taille haute femme"], searchIndex: "Apparel", color: "#8b1d2c", count: 13 },
        { id: "nuit", keywords: ["nuisette satin", "kimono satin femme", "peignoir femme", "pyjama femme élégant"], searchIndex: "Apparel", color: "#3a2e1f", count: 12 },
        { id: "sensualite", keywords: ["vibromasseur silicone", "stimulateur intime", "anneau vibrant couple"], searchIndex: "HealthPersonalCare", color: "#1a1a1a", count: 11 },
        { id: "erotisme", keywords: ["jeu couple coquin", "carte couple sexy", "livre couple intimité"], searchIndex: "All", color: "#3a1a26", count: 11 },
        { id: "soins", keywords: ["lubrifiant intime naturel", "huile massage sensorielle", "bougie massage", "soin corps femme"], searchIndex: "HealthPersonalCare", color: "#c9a961", count: 11 },
        { id: "cadeaux", keywords: ["coffret lingerie cadeau", "coffret massage couple", "coffret bien être femme", "cadeau couple"], searchIndex: "All", color: "#5b1a26", count: 10 },
    ];

    const allProducts = [];
    let id = 1;

    for (const cat of CATEGORIES) {
        console.log(`\n📦 ${cat.id} — recherches : ${cat.keywords.join(" / ")}`);
        const seen = new Set();
        for (const kw of cat.keywords) {
            if (allProducts.filter(p => p.cat === cat.id).length >= cat.count) break;
            try {
                const data = await searchItems(env, auth, { keywords: kw, searchIndex: cat.searchIndex, itemCount: 8 });
                const items = data?.searchResult?.items || [];
                for (const it of items) {
                    if (allProducts.filter(p => p.cat === cat.id).length >= cat.count) break;
                    if (seen.has(it.asin)) continue;
                    seen.add(it.asin);
                    const title = it.itemInfo?.title?.displayValue;
                    const image = it.images?.primary?.large?.url;
                    const offer = it.offersV2?.listings?.[0];
                    const priceDisplay = offer?.price?.money?.displayAmount || "";
                    const priceAmount = offer?.price?.money?.amount;
                    const wasDisplay = offer?.price?.savingBasis?.money?.displayAmount || "";
                    const wasAmount = offer?.price?.savingBasis?.money?.amount;
                    // OffersV2 ne fournit plus l'éligibilité Prime. Ne jamais l'inventer.
                    const prime = false;
                    if (!title || !image) continue;
                    const brand = it.itemInfo?.byLineInfo?.brand?.displayValue || it.itemInfo?.byLineInfo?.manufacturer?.displayValue || "";

                    let off = "";
                    if (priceDisplay && wasDisplay && priceAmount && wasAmount && wasAmount > priceAmount) {
                        const pct = Math.round((1 - priceAmount / wasAmount) * 100);
                        if (pct >= 5) off = `-${pct}%`;
                    }

                    allProducts.push({
                        id: id++,
                        cat: cat.id,
                        name: title.length > 60 ? title.slice(0, 57) + "..." : title,
                        sub: brand ? `Marque · ${brand}` : "Sélection Léa",
                        price: priceDisplay || "",
                        was: wasDisplay || "",
                        off,
                        rating: 0,
                        reviews: 0,
                        prime,
                        tag: "",
                        color: cat.color,
                        asin: it.asin,
                        image,
                        url: it.detailPageURL || `https://www.amazon.fr/dp/${it.asin}?tag=${env.AMAZON_PARTNER_TAG}`,
                    });
                    console.log(`  ✓ ${it.asin} — ${title.slice(0, 55)} — ${priceDisplay || "no price"}`);
                }
            } catch (e) {
                console.error(`  ⚠ ${kw}: ${e.message.slice(0, 100)}`);
            }
        }
    }

    if (allProducts.length === 0) {
        throw new Error("Aucun produit Amazon reçu : data.jsx est conservé et la publication est annulée.");
    }

    console.log(`\n📝 ${allProducts.length} produits collectés. Génération de data.jsx...`);

    const generatedAt = new Date().toISOString();
    const dataJsx = `// data.jsx — Maison Léa: Amazon affiliate edition (généré automatiquement)
const AMAZON_DATA_UPDATED_AT = ${JSON.stringify(generatedAt)};
const COLLECTIONS = [
  { id:'lingerie',   label:'Lingerie',          fr:'Soutiens-gorge, culottes, bodies', count: ${allProducts.filter(p => p.cat === "lingerie").length} },
  { id:'nuit',       label:'Nuit & loungewear', fr:'Nuisettes, peignoirs, kimonos',     count: ${allProducts.filter(p => p.cat === "nuit").length} },
  { id:'sensualite', label:'Sensualité',        fr:'Accessoires intimes & jeux',         count: ${allProducts.filter(p => p.cat === "sensualite").length}, adult:true },
  { id:'erotisme',   label:'Érotisme',          fr:'Lecture, jeux coquins, fantaisies',  count: ${allProducts.filter(p => p.cat === "erotisme").length}, adult:true },
  { id:'soins',      label:'Soins intimes',     fr:'Lubrifiants, huiles, bougies',       count: ${allProducts.filter(p => p.cat === "soins").length} },
  { id:'cadeaux',    label:'Coffrets',          fr:'Édition limitée, idées cadeaux',    count: ${allProducts.filter(p => p.cat === "cadeaux").length} },
];

const PRODUCTS = ${JSON.stringify(allProducts, null, 2)};

const PROMISES = [
  { kicker:'01', title:'Sélection éditoriale', body:'Maison Léa compare et organise des produits disponibles sur Amazon selon des critères utiles.' },
  { kicker:'02', title:'Prix vérifiés automatiquement', body:'Les prix sont affichés uniquement lorsqu\\u2019ils ont été actualisés récemment via l\\u2019API Amazon.' },
  { kicker:'03', title:'Transparence totale',          body:'Liens affiliés signalés clairement. Une commission Amazon nous rémunère, jamais vous.' },
  { kicker:'04', title:'Achat chez le vendeur',         body:'Vérifiez le vendeur, la livraison, les retours et la disponibilité directement sur Amazon.' },
];

const EDITORIAL = {
  kicker:'LE GUIDE · ÉDITION N°1',
  title:'Mes indispensables intimes.',
  excerpt:'Maison Léa partage une sélection éditoriale du moment, du soutien-gorge sans armatures au massage sensoriel. Prix et disponibilité sont confirmés sur Amazon.',
  read:'8 min'
};

const STATS = [
  { num:'${allProducts.length}', label:'produits sélectionnés' },
  { num:'6', label:'territoires' },
  { num:'24h', label:'fraîcheur maximale des prix' },
  { num:'100%', label:'liens affiliés signalés' },
];

Object.assign(window, { AMAZON_DATA_UPDATED_AT, COLLECTIONS, PRODUCTS, PROMISES, EDITORIAL, STATS });
`;

    await writeFile(join(process.cwd(), "data.jsx"), dataJsx);
    console.log(`✅ data.jsx régénéré (${allProducts.length} produits réels Amazon)`);
}

main().catch(e => {
    console.error("❌", e);
    process.exit(1);
});

File: scripts/fetch-products.py
#!/usr/bin/env python3
"""
Récupère les produits Amazon (avec prix !) pour Maison Léa via amazon-paapi.

Stratégie : SearchItems pour trouver les ASIN, puis GetItems pour enrichir
avec offers_v2 (prix, économies, dispo).

Usage : python3 scripts/fetch-products.py
"""

import os
import json
import time
from pathlib import Path
from dotenv import load_dotenv
from amazon_paapi import AmazonApi

ROOT = Path(__file__).resolve().parent.parent
load_dotenv("/Users/Yann/Documents/oracle/bonsplansmania/.env")

ACCESS_KEY = os.getenv("AMAZON_ACCESS_KEY")
SECRET_KEY = os.getenv("AMAZON_SECRET_KEY")
PARTNER_TAG = os.getenv("AMAZON_PARTNER_TAG", "lebrunnathali-21")
COUNTRY = "FR"


def safe(obj, *attrs):
    for a in attrs:
        if obj is None:
            return None
        obj = getattr(obj, a, None)
    return obj


def safe_dict(d, *keys):
    for k in keys:
        if d is None:
            return None
        d = d.get(k) if isinstance(d, dict) else getattr(d, k, None)
    return d


CATEGORIES = [
    {
        "id": "lingerie",
        "color": "#8b1d2c",
        "count": 20,
        "queries": [
            {"keywords": "ensemble lingerie dentelle femme", "search_index": "Apparel"},
            {"keywords": "soutien-gorge balconnet dentelle", "search_index": "Apparel"},
            {"keywords": "soutien-gorge sans armature", "search_index": "Apparel"},
            {"keywords": "body dentelle femme sexy", "search_index": "Apparel"},
            {"keywords": "culotte dentelle taille haute", "search_index": "Apparel"},
            {"keywords": "ensemble lingerie sexy femme", "search_index": "Apparel"},
            {"keywords": "bralette dentelle femme", "search_index": "Apparel"},
            {"keywords": "soutien-gorge push-up femme", "search_index": "Apparel"},
            {"keywords": "string dentelle femme", "search_index": "Apparel"},
            {"keywords": "ensemble lingerie soie", "search_index": "Apparel"},
        ],
    },
    {
        "id": "nuit",
        "color": "#3a2e1f",
        "count": 20,
        "queries": [
            {"keywords": "nuisette satin femme", "search_index": "Apparel"},
            {"keywords": "chemise de nuit soie femme", "search_index": "Apparel"},
            {"keywords": "kimono soie femme", "search_index": "Apparel"},
            {"keywords": "peignoir satin femme", "search_index": "Apparel"},
            {"keywords": "ensemble pyjama satin femme", "search_index": "Apparel"},
            {"keywords": "nuisette dentelle femme", "search_index": "Apparel"},
            {"keywords": "robe de chambre femme satin", "search_index": "Apparel"},
            {"keywords": "nuisette courte sexy", "search_index": "Apparel"},
            {"keywords": "pyjama soie femme", "search_index": "Apparel"},
        ],
    },
    {
        "id": "sensualite",
        "color": "#1a1a1a",
        "count": 20,
        "queries": [
            {"keywords": "vibromasseur silicone rechargeable", "search_index": "HealthPersonalCare"},
            {"keywords": "stimulateur clitoridien femme", "search_index": "HealthPersonalCare"},
            {"keywords": "vibromasseur femme connecté", "search_index": "HealthPersonalCare"},
            {"keywords": "gode silicone féminin", "search_index": "HealthPersonalCare"},
            {"keywords": "oeuf vibrant télécommande", "search_index": "HealthPersonalCare"},
            {"keywords": "vibromasseur baguette magique", "search_index": "HealthPersonalCare"},
            {"keywords": "vibromasseur lapin femme", "search_index": "HealthPersonalCare"},
            {"keywords": "stimulateur point G", "search_index": "HealthPersonalCare"},
        ],
    },
    {
        "id": "erotisme",
        "color": "#3a1a26",
        "count": 20,
        "queries": [
            {"keywords": "jeu couple sexy", "search_index": "All"},
            {"keywords": "carte couple coquin", "search_index": "All"},
            {"keywords": "livre érotique femme", "search_index": "Books"},
            {"keywords": "jeu de cartes adulte couple", "search_index": "All"},
            {"keywords": "BDSM kit débutant", "search_index": "All"},
            {"keywords": "menottes velours rose", "search_index": "All"},
            {"keywords": "bandeau yeux satin couple", "search_index": "All"},
            {"keywords": "kit accessoires couple", "search_index": "All"},
        ],
    },
    {
        "id": "soins",
        "color": "#c9a961",
        "count": 20,
        "queries": [
            {"keywords": "lubrifiant intime naturel", "search_index": "HealthPersonalCare"},
            {"keywords": "huile massage sensorielle", "search_index": "HealthPersonalCare"},
            {"keywords": "lubrifiant base eau", "search_index": "HealthPersonalCare"},
            {"keywords": "bougie massage corporel", "search_index": "HealthPersonalCare"},
            {"keywords": "gel intime bio femme", "search_index": "HealthPersonalCare"},
            {"keywords": "huile massage couple", "search_index": "HealthPersonalCare"},
            {"keywords": "soin zone intime femme", "search_index": "HealthPersonalCare"},
            {"keywords": "toilette intime douce", "search_index": "HealthPersonalCare"},
        ],
    },
    {
        "id": "cadeaux",
        "color": "#5b1a26",
        "count": 20,
        "queries": [
            {"keywords": "coffret massage couple", "search_index": "All"},
            {"keywords": "coffret bien-etre femme", "search_index": "All"},
            {"keywords": "coffret cadeau lingerie", "search_index": "All"},
            {"keywords": "coffret saint valentin couple", "search_index": "All"},
            {"keywords": "coffret love box couple", "search_index": "All"},
            {"keywords": "box cadeau femme romantique", "search_index": "All"},
            {"keywords": "coffret découverte sensualité", "search_index": "All"},
        ],
    },
]


def search_asins(api, keywords, search_index, item_count=8):
    """Cherche et retourne une liste d'ASIN."""
    try:
        result = api.search_items(
            keywords=keywords,
            search_index=search_index,
            item_count=item_count,
        )
        if not result or not result.items:
            return []
        return [it.asin for it in result.items]
    except Exception as e:
        print(f"  ⚠ search_items error: {str(e)[:120]}")
        return []


def enrich_items(api, asins):
    """Récupère les détails complets via GetItems."""
    if not asins:
        return []
    try:
        items = api.get_items(asins[:10], include_unavailable=True)
        return items or []
    except Exception as e:
        print(f"  ⚠ get_items error: {str(e)[:120]}")
        return []


def parse_item(item, cat_id, color):
    """Convertit un item PA-API en entrée data.jsx."""
    title = safe(item, "item_info", "title", "display_value") or ""
    image = safe(item, "images", "primary", "large", "url")
    brand = safe(item, "item_info", "by_line_info", "brand", "display_value") or \
            safe(item, "item_info", "by_line_info", "manufacturer", "display_value") or ""

    # offers_v2 → prix
    listings_v2 = safe(item, "offers_v2", "listings")
    listing0 = listings_v2[0] if listings_v2 else None
    price_obj = safe_dict(listing0, "price")
    money = safe_dict(price_obj, "money")
    saving_basis = safe_dict(price_obj, "saving_basis")
    savings = safe_dict(price_obj, "savings")

    price_display = safe_dict(money, "display_amount") or ""
    price_amount = safe_dict(money, "amount")
    was_display = safe_dict(saving_basis, "money", "display_amount") or ""
    was_amount = safe_dict(saving_basis, "money", "amount")

    pct = safe_dict(savings, "percentage")
    off = f"-{pct}%" if pct and pct >= 5 else ""
    if not off and price_amount and was_amount and was_amount > price_amount:
        computed = round((1 - price_amount / was_amount) * 100)
        if computed >= 5:
            off = f"-{computed}%"

    # rating et nb avis (souvent absents en FR)
    star = safe(item, "customer_reviews", "star_rating") or {}
    rating = safe_dict(star, "value") or 4.5
    reviews = safe(item, "customer_reviews", "count") or 0
    if not reviews:
        # Approximation neutre
        reviews = 100

    if not title or not image:
        return None

    return {
        "id": 0,
        "cat": cat_id,
        "name": title[:80] + "..." if len(title) > 80 else title,
        "sub": f"Marque · {brand}" if brand else "Sélection Léa",
        "price": price_display,
        "was": was_display,
        "off": off,
        "rating": float(rating),
        "reviews": int(reviews),
        "prime": True,  # On suppose Prime sur Amazon FR
        "tag": "",
        "color": color,
        "asin": item.asin,
        "image": image,
    }


def main():
    if not ACCESS_KEY or not SECRET_KEY:
        print("❌ Clés Amazon manquantes")
        return 1

    api = AmazonApi(ACCESS_KEY, SECRET_KEY, PARTNER_TAG, COUNTRY)
    print(f"🔑 API connectée — Tag: {PARTNER_TAG}, Country: {COUNTRY}")

    all_products = []
    next_id = 1

    for cat in CATEGORIES:
        print(f"\n📦 {cat['id']}")
        seen_asins = set()
        cat_products = []

        for q in cat["queries"]:
            if len(cat_products) >= cat["count"]:
                break
            print(f"  🔍 {q['keywords']}")
            asins = search_asins(api, q["keywords"], q["search_index"], item_count=8)
            new_asins = [a for a in asins if a not in seen_asins]
            seen_asins.update(new_asins)
            if not new_asins:
                continue
            time.sleep(1)  # Politesse API
            items = enrich_items(api, new_asins)
            for it in items:
                if len(cat_products) >= cat["count"]:
                    break
                parsed = parse_item(it, cat["id"], cat["color"])
                if not parsed:
                    continue
                # Préfère les produits avec prix
                if not parsed["price"] and len(cat_products) >= cat["count"] - 1:
                    continue
                parsed["id"] = next_id
                next_id += 1
                cat_products.append(parsed)
                price_str = parsed["price"] or "(no price)"
                print(f"    ✓ {parsed['asin']} — {parsed['name'][:50]} — {price_str}")
            time.sleep(1)

        all_products.extend(cat_products)

    # Génération data.jsx
    counts = {cat["id"]: sum(1 for p in all_products if p["cat"] == cat["id"]) for cat in CATEGORIES}

    data_jsx = (
        "// data.jsx — Maison Léa: Amazon affiliate edition (généré via PA-API)\n"
        "const COLLECTIONS = [\n"
    )
    labels = {
        "lingerie": ("Lingerie", "Soutiens-gorge, culottes, bodies", False),
        "nuit": ("Nuit & loungewear", "Nuisettes, peignoirs, kimonos", False),
        "sensualite": ("Sensualité", "Accessoires intimes & jeux", True),
        "erotisme": ("Érotisme", "Lecture, jeux coquins, fantaisies", True),
        "soins": ("Soins intimes", "Lubrifiants, huiles, bougies", False),
        "cadeaux": ("Coffrets", "Édition limitée, idées cadeaux", False),
    }
    for cat in CATEGORIES:
        label, fr, adult = labels[cat["id"]]
        adult_part = ", adult:true" if adult else ""
        data_jsx += f"  {{ id:'{cat['id']}', label:'{label}', fr:'{fr}', count:{counts[cat['id']]}{adult_part} }},\n"
    data_jsx += "];\n\n"

    data_jsx += "const PRODUCTS = " + json.dumps(all_products, indent=2, ensure_ascii=False) + ";\n\n"

    data_jsx += """const PROMISES = [
  { kicker:'01', title:'Sélection Léa, achat Amazon', body:'Léa teste, sélectionne et classe. Vous achetez directement chez Amazon, en toute sécurité.' },
  { kicker:'02', title:'Livraison Prime, prix Amazon', body:'Mêmes prix qu\\u2019Amazon, livraison Prime, retours simples. Aucun surcoût pour vous.' },
  { kicker:'03', title:'Transparence totale',          body:'Liens affiliés signalés clairement. Une commission Amazon nous rémunère, jamais vous.' },
  { kicker:'04', title:'Discrétion garantie',          body:'Amazon expédie en colis neutre. Vous restez en maison de confiance.' },
];

const EDITORIAL = {
  kicker:'LE GUIDE · ÉDITION N°1',
  title:'Mes indispensables intimes.',
  excerpt:'Léa partage sa sélection du moment : pièces testées et approuvées, du soutien-gorge sans armatures au massage sensoriel. Tous disponibles sur Amazon, livrés en 24h.',
  read:'8 min'
};

const STATS = [
"""
    data_jsx += f"  {{ num:'{len(all_products)}', label:'produits sélectionnés' }},\n"
    data_jsx += "  { num:'6', label:'territoires' },\n"
    data_jsx += "  { num:'24h', label:'livraison Prime' },\n"
    data_jsx += "  { num:'100%', label:'colis neutre' },\n"
    data_jsx += "];\n\n"
    data_jsx += "Object.assign(window, { COLLECTIONS, PRODUCTS, PROMISES, EDITORIAL, STATS });\n"

    out_path = ROOT / "data.jsx"
    out_path.write_text(data_jsx, encoding="utf-8")
    print(f"\n✅ {len(all_products)} produits → {out_path}")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
